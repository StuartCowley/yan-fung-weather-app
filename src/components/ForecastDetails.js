import React from "react";
import PropTypes from "prop-types";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  return (
    <div className="forecast-details">
      <div className="forecast-details__date">{date}</div>
      <div className="forecast-details__max-temperature">
        {temperature.min}&deg;C
      </div>
      <div className="forecast-details__max-temperature">
        {temperature.max}&deg;C
      </div>
      <div className="forecast-details__humidity">{humidity}%</div>
      <div className="forecast-details__wind">
        {wind.speed}mph {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    humidity: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
