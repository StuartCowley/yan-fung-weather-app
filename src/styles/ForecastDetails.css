.forecast-details {
  margin: 0;
  font-size: 25px;
  order: 2;
  display: grid;
  align-items: center;
  justify-content: center;
  grid-template-columns: repeat(2, 1fr);
  padding: 20px 200px;
  position: relative;
  color: rgb(36, 30, 30);
  text-shadow: 2px 2px 2px rgb(221, 215, 207);
}

.forecast-details-container-1 {
  font-size: 35px;
  background: rgb(198, 164, 164);
  opacity: 0.9;
  border-radius: 20px;
  padding-top: 20px;
  padding-bottom: 20px;
}

.forecast-details-container-2 {
  font-size: 28px;
}

.forecast-details-container-2 img {
  padding-right: 10px;
}

.forecast-details__icon {
  font-size: 40px;
}

.forecast-details__date {
  padding-bottom: 15px;
}

.forecast-details__humidity {
  padding: 2px;
}

.forecast-details__max-temperature {
  padding: 5px;
}

.forecast-details__low-temperature {
  padding: 5px;
}