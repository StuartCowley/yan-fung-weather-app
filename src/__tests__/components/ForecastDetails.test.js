import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails"

describe("ForecastDetails", () => {

  const validProps = {
   forecast: {
    date: 1111111,
    humidity: 20,
    temperature: {
      min: 10,
      max: 19,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
   },
  };

  it("render correctly", () => {
    const { asFragment } = render(
      <ForecastDetails forecast={validProps.forecast} />
    );

    expect(asFragment()).toMatchSnapshot();
  })
})