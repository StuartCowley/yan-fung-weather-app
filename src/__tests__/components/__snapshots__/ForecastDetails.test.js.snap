// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails render correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      10°C
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      19°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      20%
    </div>
    <div
      class="forecast-details__wind"
    >
      13mph s
    </div>
  </div>
</DocumentFragment>
`;
